src/main/java/org/skypro/skyshop/model/controller/ShopControllerAdvice.java

```diff
@@ -0,0 +1,17 @@
+package org.skypro.skyshop.model.controller;
+
+import org.skypro.skyshop.model.ErrorHandling.ShopError;
+import org.skypro.skyshop.model.Exceptions.NoSuchProductException;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
```

Comment:
для RestController-ов надо использовать RestControllerAdvice
```suggestion
@RestControllerAdvice
```